cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -35,6 +35,13 @@ type RestorePreamble struct {
 	MinResourceFee  int64  `json:"minResourceFee,string"`
 }
 
+// LedgerEntryDiff designates a change in a ledger entry. Before and After cannot be be omitted at the same time.
+// If Before is omitted, it constitutes a creation, if After is omitted, it constitutes a delation.
+type LedgerEntryDiff struct {
```

Comment:
I think there's value in including the type of change explicitly in a field. While it can be worked out with before / after values being null, the API is self documenting if the type is included.

For example, also including the key per the other comment:
```json
"stateDiff": [
    { "key": "ledgerkeyxdr", "type": "created", "before": null, "after": "new" },
    { "key": "ledgerkeyxdr", "type": "updated", "before": "old", "after": "new" },
    { "key": "ledgerkeyxdr", "type": "deleted", "before": "old", "after": null },
]
```

We can still use before and after so that the data is not polymorphic based on type, and for applications that just want to pull the after value, they can do that.

Reply:
I feel this is pretty redundant, but I can add it.